documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+
+                leaf-list src_port {
+                     description "List of L4 source port range as pattern '0-65365'"
+                     type string {
+                         pattern '([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])-([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])';
+                     }
+                }
+
+                leaf-list dst_port {
+                     description "List of L4 dst port range as pattern '0-65365'"
+                     type string {
+                         pattern '([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])-([0-9]{1,4}|[0-5][0-9]{4}|[6][0-4][0-9]{3}|[6][5][0-2][0-9]{2}|[6][5][3][0-5]{2}|[6][5][3][6][0-5])';
+                     }
+                }
+
+            } /* end of list DASH_ACL_RULE_LIST */
+        } /* end of container DASH_ACL_RULE */
+
+        container DASH_APPLIANCE {
+            description "DASH general appliance configs; part of APP_DB";            
+
+            list DASH_APPLIANCE_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf sip {
+                     type inet:ip-address;
+                }
+
+                leaf-list vni {
+                    type  uint32 {
+                        range 0..16777215;
+                    }
+                }
+
+            } /* end of list DASH_APPLIANCE_LIST */
+        } /* end of container DASH_APPLIANCE */
+
+        container DASH_ROUTING_TYPE {
+            description "DASH Routing types and actions; part of APP_DB";            
+
+            list DASH_ROUTING_TYPE_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern "direct|vnet|vnet-direct|drop|appliance|privatelink|privatelinknsg|servicetunnel";
+                    }
+                }
+
+                list actions {
+                    key "name"
+ 
+                    leaf name {
+                        type string {
+                            length 1..255;
+                        }
+                    }
+
+                    leaf action_type {
+                        type string {
+                            pattern "none|maprouting|direct|staticencap|appliance|4to6|mapdecap|decap|drop"
+                        }
+                    }
+
+                    leaf encap_type {
+                        type string {
+                            pattern "vxlan|nvgre"
+                        }
+                    }
+
+                    leaf vni {
```

Comment:
This indicates that all traffic with encap type vxlan use same VNI, is this correct?